src/shared/services/auth/auth.js

```diff
@@ -1,214 +1,28 @@
-
-  if (shouldRefresh) {
-    await refreshTokens();
-  }
-};
-
-const setRefreshAccessTokenInterval = () => {
-  const refreshToken = storage.getItem(REFRESH_TOKEN_KEY);
-  const decodedRefreshToken = accessTokenParser(refreshToken);
-
-  // Only start interval if refresh token is not expired
-  if (decodedRefreshToken.expiresAt * 1000 > Date.now()) {
-    setInterval(() => {
-      handleRefreshAccessToken();
-    }, ACCESS_TOKEN_REFRESH_INTERVAL);
-  }
-};
-
-/**
- * Gets the access token and return path, and clears the local storage.
- */
-function saveToken(state, accessToken) {
-  // The state param must be exactly the same as the state token we
-  // have saved in local storage (to prevent CSRF)
-  const localStateToken = storage.getItem(STATE_TOKEN_KEY);
-  const paramStateToken = decodeURIComponent(state, accessToken);
-
-  if (paramStateToken !== localStateToken) {
-    throw new Error(
-      `Authenticator encountered an invalid state token (${state}). Local state token: ${localStateToken}.`
-    );
-  }
-
-  tokenData = accessTokenParser(accessToken);
-  const localNonce = storage.getItem(NONCE_KEY);
-  if (tokenData.nonce && tokenData.nonce !== localNonce) {
-    throw new Error(
-      `Authenticator encountered an invalid nonce (${tokenData.nonce}). Local nonce token: ${localNonce}.`
-    );
-  }
-
-  storage.setItem(ACCESS_TOKEN_KEY, accessToken);
-
-  storage.removeItem(STATE_TOKEN_KEY);
-  storage.removeItem(NONCE_KEY);
-
-  // Clean up URL; remove query and hash
-  // https://stackoverflow.com/questions/4508574/remove-hash-from-url
-  global.history.replaceState('', document.title, global.location.pathname);
-}
-
-/**
- * Fetch tokens with given authorization code.
- */
-async function fetchTokensByCode(code) {
-  const { authEndpoint, realm, clientId } = configuration.keycloak;
-
-  const searchParams = new URLSearchParams({
-    code,
-    grant_type: 'authorization_code',
-    client_id: clientId,
-    redirect_uri: AUTH_REDIRECT_URI,
-  });
-
-  const tokenUrl = `${authEndpoint}/realms/${realm}/protocol/openid-connect/token`;
-
-  const response = await fetch(tokenUrl, {
-    body: searchParams.toString(),
-    method: 'POST',
-    headers: { 'Content-type': 'application/x-www-form-urlencoded' },
-  });
-
-  return response.json();
+export function getOauthDomain() {
+  return storage.getItem(OAUTH_DOMAIN_KEY);
 }
 
 /**
- * Handle login callback.
+ * Returns auth implementation.
  */
-async function handleAuthorizationCallback() {
-  // Parse query string into object
-  const params = queryStringParser(global.location.hash);
-
-  if (!params || !params.state) return;
-
-  if (params.code) {
-    // Handle keycloak callback
-    const tokens = await fetchTokensByCode(params.code);
-    saveToken(params.state, tokens.access_token);
-    saveRefreshToken(tokens.refresh_token);
-    return;
+export function getAuth() {
+  if (configuration.keycloak && getOauthDomain() === 'keycloak') {
```

Comment:
Use ternary expression here.

Reply:
Why no function expressions?

Reply:
What do you mean with function expressions in this case?

Reply:
`export const getAuth = () => configuration.keycloak && getOauthDomain() === 'keycloak' ? keycloak : authz;`

Reply:
You're right - I've used function here because it was line with code already in place. Will refactor to arrow functions. 

(I was confused because your comment says 'why no function expressions' while function expressions are being used)